Tests/XcodeGenKitTests/ProjectGeneratorTests.swift

```diff
@@ -1004,4 +1004,38 @@ class ProjectGeneratorTests: XCTestCase {
             }
         }
     }
+    
+    func testGenerateXcodeProjectWithDestination() throws {
+        let groupName = "App_iOS"
+        let sourceDirectory = fixturePath + "TestProject" + groupName
+        let frameworkWithSources = Target(
+            name: "MyFramework",
+            type: .framework,
+            platform: .iOS,
+            sources: [TargetSource(path: sourceDirectory.string)]
+        )
+        
+        describe("generateXcodeProject") {
+            $0.context("without projectDestinationDirectory") {
+                $0.it("generate groups") {
+                    let project = Project(name: "test", targets: [frameworkWithSources])
+                    let generator = ProjectGenerator(project: project)
+                    let generatedProject = try generator.generateXcodeProject()
+                    let group = generatedProject.pbxproj.groups.first(where: { $0.nameOrPath == groupName })
+                    try expect(group?.path) == "App_iOS"
```

Comment:
This value should be relative path from source root.
https://github.com/yonaskolb/XcodeGen/pull/681/files#r332485507